cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/partitioned/PartitionConsumer.java

```diff
@@ -55,4 +56,13 @@
    * @param succeeded whether or not processing of the specified partitions was successful
    */
   void onFinish(List<? extends Partition> partitions, boolean succeeded);
+
+  /**
+   * Same as {@link #onFinish(List, boolean)}, but allows specifying {@link PartitionKey}s
+   * instead of {@link co.cask.cdap.api.dataset.lib.Partition}s.
+   *
+   * @param partitionKeys list of partition keys to mark as either succeeded or failed processing
+   * @param succeeded whether or not processing of the specified partitions was successful
+   */
+  void onFinishWithKeys(List<? extends PartitionKey> partitionKeys, boolean succeeded);
```

Comment:
I think its fine to take in a List, because the user will pass in the list from the object they get in `consumePartitions` method, which returns them a List.
So they will already have access to any list to pass in (in the case of the similar `onFinish(List<? extends Partitions> partitions, ...` method). It also keeps the two onFinish methods consistent.